### DISCUSSION TITLE
Allow multiple roles per user

### DISCUSSION THREAD

**Discussion by HashemKhalifa** [directus/directus#2741] [Feature Requests]
Is there are any plans to have multiple roles for one user?

**Comment by benhaynes** [comment]
No — we thought about it before, but are moving to a "Dynamic Permissions" system instead. So each user will only have one role, but you'll be able to create as many granular roles as you need.

**Reply by kees-tce** [reply]
Is there already such dynamic permission in place?

And how can I implement this use case:
- users have a role eg. Marketeer, admin, customer, contractor etc 
- a subset of all users, regardless of the role above, should also have a role "webshop user" which will be used on the frontend of the webshop and check if the user is authorized to access the webshop

Any thoughts on this?

**Reply by rijkvanzanten** [reply]
If it's only for your end websites use, you could add a custom dropdown field (or a m2o) to directus users and use that :)

**Comment by rijkvanzanten** [comment]
Duplicate discussion https://github.com/directus/directus/discussions/4959 took over all the eyeballs on this discussion. We'll keep using that to track the popularity of this feature 🙂
